cli/src/lib/formatAppAuthName.js

```diff
@@ -0,0 +1,30 @@
+const APP_AUTH_PREFIX = 'M_'
+const DHIS_WEB = 'dhis-web-'
+
+/**
+ * Returns the string that identifies the 'App view permission'
+ * required to view the app
+ *
+ * Ex: coreApp && name = 'data-visualizer': authName = 'M_dhis-web-data-visualizer'
+ * Ex: name = 'pwa-example': authName = 'M_pwaexample'
+ * Ex: name = 'BNA Action Tracker': authName = 'M_BNA_Action_Tracker'
+ */
+const formatAppAuthName = (config) => {
+    if (config.coreApp) {
+        // TODO: Verify this formatting - are there any transformations,
+        // or do we trust that it's lower-case and hyphenated?
+        return APP_AUTH_PREFIX + DHIS_WEB + config.name
+    }
```

Comment:
Curious about this formatting

Reply:
We do need to do transformation here, as the "name" in the d2 config is NOT necessarily what's used for the authority (or URL name) - see [here](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/App.java#L494-L499).  We already replicate this logic in the platform [here](https://github.com/dhis2/app-platform/blob/master/cli/src/lib/constructAppUrl.js)

Reply:
Ah I see we're doing that below - but we should also do it for core apps.

Reply:
Yeah, notice that the formatting is different between those two things you linked, namely around hyphens -- core apps seem to be getting a different treatment?

Reply:
Added some formatting insurance